src/nvim/ex_cmds.c

```diff
@@ -5935,3 +6036,191 @@ void set_context_in_sign_cmd(expand_T *xp, char_u *arg)
     }
   }
 }
+
+/// Open a window for future displaying of the inc_sub mode.
+///
+/// Does not allow editing in the window.
+/// Returns when the window is closed.
+///
+/// @param pat    The pattern word
+/// @param sub    The replacement word
+/// @param lmatch The list containing our data
+void ex_window_inc_sub(char_u * pat,
```

Comment:
I believe this function is responsible for trigger the right redraws even when a window is not opened `incsub=nosplit`. Maybe `static void inc_sub_display`.